Xamarin.Forms.Core/Storyboard.cs

```diff
@@ -0,0 +1,551 @@
+		//public static void SetTargetProperty(this Storyboard s, string property);
+	}
+
+	public interface IInterpolatable
+	{
+		object InterpolateTo(object from, object target, double interpolation);
+	}
+
+	public interface IAnimation
+	{
+		void Begin();
+	}
+
+	public interface IAnimation<T> : IAnimation
+	{
+		T From { get; set; }
+		T To { get; set; }
+	}
+
+	public interface IInterpolatable<T> : IInterpolatable
+	{
+		T InterpolateTo(T from, T target, double interpolation);
+	}
+
+	public abstract class Animation<T> : Timeline
+	{
+		public abstract T From { get; set; }
+		public abstract T To { get; set; }
+	}
+
+
+	public class AnimateExtension : BindableObject, IMarkupExtension<Timeline>
+	{
+
+		public BindableProperty Property { set; get; }
+
+		public string Binding { set; get; }
+
+		public Timeline ProvideValue(IServiceProvider serviceProvider)
+		{
+			var anim = new PositionAnimation();
+
+			BindingBase GetBinding()
+			{
+				return new Binding(Binding);
+			}
+
+			anim.SetBinding(PositionAnimation.ToProperty, GetBinding());
+			Storyboard.SetTargetProperty(anim, Property);
+			return anim;
+		}
+
+		object IMarkupExtension.ProvideValue(IServiceProvider serviceProvider)
+		{
+			return (this as IMarkupExtension<Timeline>).ProvideValue(serviceProvider);
+		}
+	}
+
+
+	public abstract class InterpolatableAnimation<T> : Animation<T> where T : IInterpolatable<T>
+	{
+	}
+
+
+	public class PositionAnimation : Animation<int>
+	{
+
+		public static readonly BindableProperty FromProperty = BindableProperty.Create(nameof(From), typeof(int), typeof(PositionAnimation), 0);
+
+		public static readonly BindableProperty ToProperty = BindableProperty.Create(nameof(To), typeof(int), typeof(PositionAnimation), 0);
+
+		public override int From
+		{
+			get { return (int)GetValue(FromProperty); }
+			set { SetValue(FromProperty, value); }
+		}
+
+		public override int To
+		{
+			get { return (int)GetValue(ToProperty); }
+			set { SetValue(ToProperty, value); }
+		}
+
+	}
+
+	public class DoubleAnimation : Animation<double>
+	{
+		public static readonly BindableProperty FromProperty = BindableProperty.Create(nameof(From), typeof(double), typeof(DoubleAnimation), .0);
+
+		public static readonly BindableProperty ToProperty = BindableProperty.Create(nameof(To), typeof(double), typeof(DoubleAnimation), .0);
+
+		public override double From
+		{
+			get { return (double)GetValue(FromProperty); }
+			set { SetValue(FromProperty, value); }
+		}
+
+		public override double To
+		{
+			get { return (double)GetValue(ToProperty); }
+			set { SetValue(ToProperty, value); }
+		}
+	}
+
+	public abstract class Timeline : BindableObject, IAnimation
+	{
+		const uint defaultDuration = 300;
+
+		protected Timeline()
+		{
+			Duration = defaultDuration;
+			Easing = Easing.SpringOut;
+		}
+
+		public uint BeginTime { get; set; }
+
+		public uint Duration { get; set; }
+
+		public bool Reverse { get; set; }
+
+		public bool Loop { get; set; }
```

Comment:
How about something like [RepeatBehavior](https://docs.microsoft.com/es-es/dotnet/api/system.windows.media.animation.timeline.repeatbehavior?view=netframework-4.8)? With the bool type, we launch the animation 1 time, or in Loop. However, RepeatBehavior could allow to repeat the animation also a certain number of times (Example: 3 times).